### PR TITLE
Added Azure Signals Object to Strategy Component

### PR DESCRIPTION
**Pull request by IGUDE2**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : No
- I have compiled the code successfully my laptop : No
- I have written examples to demonstrates the usage of the newly implemented APIs : No
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/api/azure/elastiGroup/V3/ApiSignalsAzure.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiSignalsAzure {
+
+    // Partial Update support
+    private Set<String> isSet;
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/elastiGroup/V3/ApiSignalsAzure.java:17]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/api/azure/elastiGroup/V3/ApiSignalsAzure.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.enums.ElastigroupVmSignalEnumAzure;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiSignalsAzure {
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/elastiGroup/V3/ApiSignalsAzure.java:13]: implements IPartialUpdateEntity

File: src/main/java/com/spotinst/sdkjava/model/api/azure/elastiGroup/V3/ApiSignalsAzure.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.enums.ElastigroupVmSignalEnumAzure;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/elastiGroup/V3/ApiSignalsAzure.java:12]: add @JsonFilter("PartialUpdateEntityFilter")

File: src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/ElastigroupConverterAzure.java [hunk 4]
```diff
@@ -364,6 +364,13 @@ private static ApiStrategyAzure toDal(ElastigroupStrategyConfigurationAzure stra
             if (strategy.isOrientationSet()) {
                     retVal.setOrientation(strategy.getOrientation().getName());
             }
+            if(strategy.isSignalsSet()) {
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/ElastigroupConverterAzure.java:368]: remove extra line space

File: src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/SignalsAzure.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
+package com.spotinst.sdkjava.model.bl.azure.elastiGroup.V3;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.spotinst.sdkjava.enums.ElastigroupVmSignalEnumAzure;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/SignalsAzure.java:9]: add these
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)

File: src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/SignalsAzure.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        return type;
+    }
+
+    public void setType(ElastigroupVmSignalEnumAzure type) {
+        isSet.add("type");
+        this.type = type;
+    }
+
+    //region Object overrides
+    @Override
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/SignalsAzure.java:53]: why do we need override methods in BL class

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/SignalsAzure.java (hunk 6 above)
Comment (IGUDE2) [reply src/main/java/com/spotinst/sdkjava/model/bl/azure/elastiGroup/V3/SignalsAzure.java]: Removed the override methods in BL Class and Tested, call working fine

**Review by chandra1-n — APPROVED** [review]
no changes required

**Pull request merged**